control/available_plugin.go

```diff
@@ -442,43 +445,24 @@ func (p *apPool) kill(id uint32, reason string) {
 	}
 }
 
-// kills the plugin with the lowest hit count
-func (p *apPool) killLeastUsed(reason string) {
-	p.Lock()
-	defer p.Unlock()
-
-	var (
-		id   uint32
-		prev int
-	)
-
-	// grab details from the first item
-	for _, p := range p.plugins {
-		prev = p.hitCount
-		id = p.id
-		break
-	}
-
-	// walk through all and find the lowest hit count
-	for _, p := range p.plugins {
-		if p.hitCount < prev {
-			prev = p.hitCount
-			id = p.id
-		}
+func (p *apPool) selectAndKill(taskID, reason string) {
+	sp := make([]strategy.SelectablePlugin, p.count())
```

Comment:
Refactor this method so that everything (select and kill) is done inside the strategies remove method.